src/advanced_unsafety/invalid_values.md

```diff
@@ -34,6 +34,12 @@ It's also currently invalid for `Vec<T>` to have a null pointer for its buffer!
 
 There are a lot of other reasons that a pointer type may not be valid, but these are the ones where the bit pattern is statically known to be invalid regardless of context. We'll be covering these in more depth in other chapters (@@note: where?), but, for example, all of these pointers must not only be non-null, they must also point to an actual valid instance of that type at all times (except `Vec<T>`, which is allowed to refer to invalid-but-aligned-and-non-null memory when it is empty)
 
+#### "shallow" vs "deep" validity
+
+
+An open question in Rust's model is whether references and reference-like types have "shallow" validity (roughly, the rules above), or "deep" validity (where a reference is valid only when the pointed-to data is valid, and that applies transitively). This issue is tracked upstream as [UGC #77](https://github.com/rust-lang/unsafe-code-guidelines/issues/77). The current discussion seems to skew towards shallow validity as opposed to deep validity, but this code change.
```

Comment:
Consider "referent" or "pointee" over "pointed-to data", depending on whether you want to avoid using jargon or avoid confusion over what "data" means 😉

Reply:
This is explicitly learning materials, so we're trying not to use too much jargon